src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -292,6 +292,9 @@ int IDebugPendingBreakpoint2.Bind()
             {
                 if (e.GetBaseException() is InvalidCoreDumpOperationException)
                     return AD7_HRESULT.E_CRASHDUMP_UNSUPPORTED;
+                else if (e.GetBaseException() is ArgumentException && (enum_BP_LOCATION_TYPE)_bpRequestInfo.bpLocation.bpLocationType == enum_BP_LOCATION_TYPE.BPLT_DATA_STRING)
+                    // handle invalid data bp address/size
+                    return Constants.E_FAIL;
```

Comment:
I don't think this is where you want this - if binding takes a while we will give up synchronisouly waiting.

Reply:
You probably want BindAsync to return a Task<int> instead. Or maybe you don't want to return E_FAIL at all but rather raise a breakpoint errer event.

Reply:
Yup, worked it out in order to raise a breakpoint error event (but also still return E_FAIL so that the caller won't return an invalid address)